### PR TITLE
athena audit logs - use otel traces in querier

### PR DESCRIPTION
**Pull request by tobiaszheller**
Part of https://github.com/gravitational/teleport.e/issues/894
RFD: https://github.com/gravitational/teleport/blob/master/rfd/0118-scalable-audit-logs.md

Since we have ctx now in Seach interface propagated, we can use tracing in athena logger.

### CODE REVIEW

**Review by rosstimothy — COMMENTED** [review]
If you wanted to further extend tracing to the [aws client](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws/example/v0.42.0/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws/example/main.go#L62-L68) we could leverage [otelaws](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws)

File: lib/events/athena/querier.go [hunk 1]
```diff
@@ -309,6 +339,8 @@ func prepareQuery(params searchParams) (query string, execParams []string) {
 }
 
 func (q *querier) startQueryExecution(ctx context.Context, query string, params []string) (string, error) {
+	ctx, span := q.tracer.Start(ctx, "startQueryExecution")
```
[the remark this answers is not in the document]
Comment (tobiaszheller) [reply lib/events/athena/querier.go]: I have added athena prefix and added query_id as attribute. From, limit and to are used as attributes on top level call. https://github.com/gravitational/teleport/pull/26826/commits/802c2972cd231d6a736e8fad8ecfd28ae99dbdb0

File: lib/events/athena/querier.go [hunk 2]
```diff
@@ -114,6 +124,16 @@ func newQuerier(cfg querierConfig) (*querier, error) {
 }
 
 func (q *querier) SearchEvents(ctx context.Context, req events.SearchEventsRequest) ([]apievents.AuditEvent, string, error) {
+	ctx, span := q.tracer.Start(
+		ctx,
+		"events/athena/SearchEvents",
```
[the remark this answers is not in the document]
Comment (tobiaszheller) [reply lib/events/athena/querier.go]: i went with `audit/SearchEvents`, and as `otel.library.name` we will set `athena`.

**Review by timothyb89 — APPROVED** [review]

**Comment by tobiaszheller** [comment]
@gabrielcorado @rosstimothy friendly ping, PTAL :)

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**